components/action-mgt/org.wso2.carbon.identity.action.execution/src/main/java/org/wso2/carbon/identity/action/execution/impl/ActionExecutorServiceImpl.java

```diff
@@ -312,18 +316,16 @@ private ActionExecutionStatus processActionResponse(Action action,
         } else if (actionInvocationResponse.isError() && actionInvocationResponse.getResponse() != null) {
             return processErrorResponse(action, (ActionInvocationErrorResponse) actionInvocationResponse.getResponse(),
                     eventContext, actionRequest, actionExecutionResponseProcessor);
-        } else {
-            logErrorResponse(action, actionInvocationResponse);
         }
-
-        return new ActionExecutionStatus(ActionExecutionStatus.Status.FAILED, eventContext);
+        logErrorResponse(action, actionInvocationResponse);
+        throw new ActionExecutionRuntimeException("Received an invalid or unexpected action status.");
```

Comment:
Add more context info like action type, action id

Reply:
This is fixed with f433c295103c3f59e98452eb08136a24f978bdd3